Source/WebCore/loader/FrameLoader.cpp

```diff
@@ -1290,6 +1290,10 @@ void FrameLoader::loadInSameDocument(URL url, RefPtr<SerializedScriptValue> stat
                 domWindow->protectedNavigation()->updateForNavigation(*currentItem, navigationType);
             }
         }
+
+        // During the NavigateEvent the page could have been detached from JS.
+        if (!m_frame->page())
+            return;
```

Comment:
Is this check needed now or also in the past? Any test that used tobe affected?

Reply:
It is still needed as the FrameLoader accesses the page after this point.